assets/thanos-querier/deployment.yaml

```diff
@@ -80,59 +80,45 @@ spec:
         - mountPath: /etc/tls/grpc
           name: secret-grpc-tls
       - args:
-        - -provider=openshift
-        - -https-address=:9091
-        - -http-address=
-        - -email-domain=*
-        - -upstream=http://localhost:9090
-        - -openshift-service-account=thanos-querier
-        - '-openshift-sar={"resource": "namespaces", "verb": "get"}'
-        - '-openshift-delegate-urls={"/": {"resource": "namespaces", "verb": "get"}}'
-        - -tls-cert=/etc/tls/private/tls.crt
-        - -tls-key=/etc/tls/private/tls.key
-        - -client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token
-        - -cookie-secret-file=/etc/proxy/secrets/session_secret
-        - -openshift-ca=/etc/pki/tls/cert.pem
-        - -openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
-        - -bypass-auth-for=^/-/(healthy|ready)$
-        env:
-        - name: HTTP_PROXY
-          value: ""
-        - name: HTTPS_PROXY
-          value: ""
-        - name: NO_PROXY
-          value: ""
-        image: quay.io/openshift/oauth-proxy:latest
+        - --secure-listen-address=0.0.0.0:9091
+        - --upstream=http://127.0.0.1:9090
+        - --config-file=/etc/kube-rbac-proxy/config.yaml
+        - --tls-cert-file=/etc/tls/private/tls.crt
+        - --tls-private-key-file=/etc/tls/private/tls.key
+        - --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
+        - --logtostderr=true
+        - --ignore-paths=/-/ready,/-/healthy
+        image: quay.io/brancz/kube-rbac-proxy:v0.11.0
         livenessProbe:
           failureThreshold: 4
           httpGet:
             path: /-/healthy
-            port: 9091
+            port: web
             scheme: HTTPS
           initialDelaySeconds: 5
           periodSeconds: 30
-        name: oauth-proxy
+        name: kube-rbac-proxy-web
         ports:
         - containerPort: 9091
           name: web
         readinessProbe:
           failureThreshold: 20
           httpGet:
             path: /-/ready
-            port: 9091
+            port: web
             scheme: HTTPS
           initialDelaySeconds: 5
           periodSeconds: 5
         resources:
           requests:
             cpu: 1m
-            memory: 20Mi
+            memory: 15Mi
```

Comment:
i would leave the requests as-is

Reply:
OK, I will change it to 20MiB. 
From which criteria do we know this kube-rbac-proxy should require 20MiB memory while its neighbor container kube-rbac-proxy for tenancy requires 15MiB? 🤔

Reply:
https://github.com/openshift/enhancements/blob/master/CONVENTIONS.md#resources-and-limits describes how to determine resource requests for OpenShift components. The other kube-rbac-proxy is only for serving /metrics while this container would serve the /api/v1/* requests so I would expect that the latter uses more resources. I think it is wiser to leave the requests untouched for now and check later if we can decrease the values.